src/main/kotlin/com/demonwav/mcdev/creator/CreatorStep.kt

```diff
@@ -0,0 +1,125 @@
+import com.intellij.openapi.application.runReadAction
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.SmartPsiElementPointer
+import com.intellij.psi.util.createSmartPointer
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption.CREATE
+import java.nio.file.StandardOpenOption.TRUNCATE_EXISTING
+import java.nio.file.StandardOpenOption.WRITE
+
+/**
+ * Represents a discrete kind of configuration code for a project. Project creators use these to spread implementation
+ * out between platforms without coupling them, and allow overriding for more complex options for specific build systems
+ * without coupling the build system and the platform.
+ */
+interface CreatorStep {
+
+    fun runStep(indicator: ProgressIndicator)
+
+    companion object {
+        private val scheduledReformats: MutableList<SmartPsiElementPointer<PsiFile>> = mutableListOf()
+        fun runAllReformats() {
+            runWriteTask {
+                for (scheduledReformat in scheduledReformats) {
+                    val file = scheduledReformat.element ?: continue
+                    ReformatCodeProcessor(file, false).run()
+                }
+            }
+            scheduledReformats.clear()
+        }
+
+        fun writeTextToFile(
+            project: Project,
+            targetDir: Path,
+            fileName: String,
+            text: String
+        ): VirtualFile {
+            if (Files.notExists(targetDir)) {
+                Files.createDirectories(targetDir)
+            }
+            val file = targetDir.resolve(fileName)
+            Files.write(file, text.toByteArray(Charsets.UTF_8), WRITE, CREATE, TRUNCATE_EXISTING)
+            val vFile = file.virtualFileOrError
+
+            // Reformat the code to match their code style
+            runReadAction {
+                PsiManager.getInstance(project).findFile(vFile)?.let {
+                    scheduledReformats += it.createSmartPointer()
+                }
+            }
+
+            return vFile
+        }
+
+        fun writeText(file: Path, text: String, psiManager: PsiManager? = null) {
+            Files.write(file, text.toByteArray(Charsets.UTF_8), CREATE, TRUNCATE_EXISTING, WRITE)
+            psiManager?.findFile(file.virtualFileOrError)?.let {
+                ReformatCodeProcessor(it, false).run()
+            }
+        }
+    }
+}
+
+class BasicJavaClassStep(
+    private val project: Project,
+    private val buildSystem: BuildSystem,
+    private val className: String,
+    private val classText: String
+) : CreatorStep {
+    override fun runStep(indicator: ProgressIndicator) {
+        val dirs = buildSystem.dirsOrError
+
+        runWriteTask {
+            indicator.text = "Writing main class"
```

Comment:
```suggestion
            indicator.text = "Writing class: $className"
```
That's the main one (no pun intended), but there's a couple of things in this class that need updating

Reply:
Thanks, good catch

Reply:
Finally got around to addressing this, should be good now. This PR also now includes `.gitignore` files.